src/astarte-client/types/dataType.ts

```diff
@@ -16,4 +16,20 @@
    limitations under the License.
 */
 
-export type AstarteDataType = number | boolean | string | number[] | boolean[] | string[];
+export type AstarteDataValue = number | boolean | string | number[] | boolean[] | string[];
```

Comment:
Could you elaborate on this? Do you mean we should have just a tuple type, to enforce a relationship between the data `type` and `value`?

Reply:
you need a specific type for the labels, as they are used in mappings too. So it's fine to have AstarteDataType as a standalone type.
On the other hand, AstarteDataValue cannot be just a union of some js types, as this will not tell you what's inside of it.
is `string` representing a date, a binary or just a plain string? you need to couple value and type for that. But if you couple DataValue and DataType in an object, you're not using the compiler to check whether they are actually matching or not.

When reading something like `AstarteBinaryValue[]` the typescript compiler will know for sure it's an array for string values, each one representing a binary value

Reply:
However Typescript only does static checks. `AstarteDataValue` is used when fetching or parsing `AstarteInterfaceValues`, and I cannot see another way to label javascript type returned from the API. That, unless we decide to only return parsed values from `astarte-client`; but I guess this is out of scope here.

Reply:
this is a middle ground. It's on us to decide where to draw the boundary. It's true, it you start changing this check in the astarte-client, this PR becomes too big, but I think this is also necessary to properly design the chart api.
Since we planned this change already (returning typed astarte values from the client) we could rebase this PR ontop of that feature, which on a side note, will also benefit the live-events handing